### PR TITLE
Clean Gradle config, move to Lyricist and AndroidX ViewModel, clean ViewModel code, add api text files

### PR DESCRIPTION
**Pull request by GerardPaligot**
## Changelog

* Upgrade Kotlin to 2.0.0
* Migrate translation from Moko to Lyricist
* Migrate ViewModel from Moko to AndroidX ViewModel
* Clean Gradle dependencies
* Fix a crash when OpenFeedback return an empty comment
* Create a new module for ui models shared between viewmodel and m3 modules.
* Move OpenFeedback component from m3 to viewmodel modules.
* m3 module doesn't know anymore viewmodel and openfeedback modules.
* Reduce the complexity of the ViewModel class.
* Add api text files for every module

### CODE REVIEW

File: gradle.properties
Comment (martinbonnin) [inline gradle.properties]: FWIW, we can remove that now. It was an issue with bintray or sonatype but it's been fixed now. See https://github.com/gradle/gradle/issues/11308

**Review by martinbonnin — APPROVED** [review]

File: gradle.properties
Comment (GerardPaligot) [reply gradle.properties]: Are you sure? I can only see people who said that they still need to add this parameter 🤔

**Comment by GerardPaligot** [comment]
Not sure what is the issue with the CI build 🤔

File: gradle.properties
Comment (martinbonnin) [reply gradle.properties]: Yup, all good. See https://github.com/gradle/gradle/issues/11412#issuecomment-733655244

Comment (martinbonnin) [reply gradle.properties]: I removed it in https://github.com/paug/openfeedback-sdk-kotlin/pull/29/commits/e22400f42b9d47fd7f166a920bd222e8c921505f

Comment (GerardPaligot) [reply gradle.properties]: A comment not easy to find! :D

**Comment by GerardPaligot** [comment]
Added a new commit to avoid a crash when we try to display an empty comment.

File: gradle.properties
Comment (martinbonnin) [reply gradle.properties]: Yeaaaaaaaa, it's a secret handshake between lib developers :P

**Comment by martinbonnin** [comment]
> Not sure what is the issue with the CI build 🤔

~It's (again) a Gradle classloader issue~ Edit: I changed my mind, it's a bouncy castle issue where they publish incompatible code at two distinct coordinates. You can reproduce locally by removing your debug keystore: `rm ~/.android/debug.keystore`. 

BouncyCastle has the same classes at 2 distinct coordinates: `-jdk15on.jar` and `-jdk18on.jar`. I'll make a fix but feel free to merge this one for the time being.

**Review by martinbonnin — APPROVED** [review]

**Comment by GerardPaligot** [comment]
Gradle is cool when it works and a nightmare when you need a BAC +15 to fix internal issue...

No emergency in the merge of this contribution and I would like to work on the public API of this library:

* An UI module without the domain/data layer
* Be able to load votes and/or comments from a custom ViewModel (and provide a default one)
* If the viewmodel is no more injected in the composable, we should be able to avoid the initialization step in the application class

**Comment by GerardPaligot** [comment]
Well, done: https://github.com/paug/openfeedback-sdk-kotlin/pull/29#issuecomment-2211680260 ^^

**Comment by martinbonnin** [comment]
> Gradle is cool when it works and a nightmare when you need a BAC +15 to fix internal issue...

Gradle is just messenger here. This is "just" regular dependency hell and I think I blame the BouncyCastle maintainers. Had they released a [multi-release jar file](https://www.baeldung.com/java-multi-release-jar) instead of duplicating the maven coordinates, I think this problem wouldn't show 🤷 .

edit: but of course multi-release doesn't work with Java 5 so 🤷

**Comment by martinbonnin** [comment]
Workaround here: https://github.com/paug/openfeedback-sdk-kotlin/pull/30/files

**Comment by GerardPaligot** [comment]
Rebase done. Probably a good idea to publish a SNAPSHOT version first with all these changes and when it'll be ok, publishing a major version (1.0.0-alpha01 or 1.0.0-beta01 if we want)?

**Comment by martinbonnin** [comment]
> Rebase done. Probably a good idea to publish a SNAPSHOT version first with all these changes and when it'll be ok

If I made my job right, SNAPSHOTs are published on every push to `main` so there should be one done just after this PR is merged

> publishing a major version (1.0.0-alpha01 or 1.0.0-beta01 if we want)?

Woohoo, 1.0.0 is coming 🎉 !! Yup, if we feel good, let's go for it. Probably do a `1.0.0-alpha.1` first.

**Comment by GerardPaligot** [comment]
Oh, it means I don't need to do this: https://github.com/paug/openfeedback-sdk-kotlin/commit/55ce6ae74810112b66044e98bc552320b45baa6f ?

**Comment by martinbonnin** [comment]
Sorry you definitely need that. In theory, `main` should always have a `-SNAPSHOT` version....

**Pull request merged**
